### DISCUSSION TITLE
<FIBOS-WORMHOLE>FIBOS 生态激励提案(基础设施类)

### DISCUSSION THREAD

**Discussion by BIU64** [FIBOSIO/fibos#25] [General]
Hello FIBOSers,上周开发者委员会的成员们确定了一个生态激励机制的初版(https://github.com/FIBOSIO/fibos/discussions/23#discussioncomment-1393602)，作为该版提案的起草者之一，衷心期待该方案可以成为 FIBOS 持续发展的一颗种子！
提案通过后，我们也讨论了很多关于“FIBOS 到底需要什么”的问题，有效的行业痛点工具？联通 FIBOS 和其他公链的开发基础设施？更 FOMO 的生态项目？毫无疑问，前述的方面都是 FIBOS 需要的，但在 FO 经历了短期内的快速波动后，结合流动性薄弱和社区发展瓶颈的现状，我们认为可以先从 FIBOS 与其他公链的交互联通开始着手。
因此，CryptoAboriginal(CA) 在此提出了以下方案，期待能够从用户端构建主流 DeFi 用户和 FIBOS 的关联，使得现有的生态工具都能够被真正的用起来，最大化的降低用户的使用成本和尝试畏难情绪。

**提案名称：FIBOS-WORMHOLE 提案**
Cross-chain interoperable middleware(跨链交互技术)

_提案背景_
FIBOS-WORMHOLE(FWH)实现了任意 ETH 同构公有区块链用户对 FIBOS 区块链的跨链可操作性。

目前，FIBOS 已经实现了与 Ethereum 主链的异构去中心化跨链，与其他同构链的异构跨链因此具备了同样的实现基础。但 FIBOS 无法被 MetaMask、Math 等主流 DeFi 工具兼容，从而导致第三方用户使用 FIBOS 链上应用的学习成本增加。

_提案介绍_
Wormhole 方案将解决上述问题。借助 FWH 工具，开发者可以实现智能合约部署在 FIBOS 链上、但任意 ETH 同构公链用户均可直接通过已有的 Metamask 等插件直接跨链交互，同时不损失安全性和去中心化属性。
依靠 FWH，主要的 DeFi、GameFi 玩家们无需安装额外插件或创建私钥对，仅需使用已有插件即可得到更顺畅、更低成本的体验。

同时，ETH 宇宙的 Layer2 浪潮证明了行业和市场对于公链性能拓展的真实需求，而 FIBOS 已有的基础设施和方案完全具有足够的能力与各通用型 Layer2 方案竞争。Wormhole 方案将彻底打通 FIBOS 作为 ETH 宇宙通用型 Layer2 的阻碍。

总而言之，我们认为 Wormhole 方案的实现将主要在以下方面为 FIBOS 生态带来深层价值：
1.主流 DeFi 用户的 0 使用成本和学习成本，以及极佳的链上体验。
2.开发者能够真正兼顾以太坊用户和 FIBOS 的开发、运营低成本。
3.FIBOS 将能够探索多链通用型 Layer2、多链资产跨链 HUB、DeFi 资产汇集、NFT、GameFI、DAO 等方面应用。 

同时，本提案具有以下优势：
1. 用户使用成本降低和体验提升
2. 开发成本降低且不影响用户端
3. 增加 FIBOS 链上用户、资金往来交互
4. FIBOS 将成为所有 ETH 同构链的异构 Layer2

_技术实现方案_ 
Wormhole 方案实际上承担了“翻译官”的作用，将不同公链公钥格式以及密码签名格式转读为 FIBOS 公链可兼容的格式，该方案力图直接对公钥以及签名内容进行转录，过程中无需用户私钥参与，能够最大程度上保护用户的私钥安全，配合 FIBOS 与 ETH 的资产异构跨链，使得跨链交互成为可能。

FWH 的实现依赖如下 3 个方面：
密码学：通过大数计算等数学方法完成以太坊格式的公钥以及签名格式到FIBOS兼容的格式转换
FIBOS 异构跨链：依赖 FIBOS 异构跨链，完成以太坊与FIBOS链上的资产跨入与跨出，同时扩展 FIBOS 异构跨链协议，支持基础合约跨链调用
RPC转换中间件：为了能够通过 MetaMask 完成 FIBOS 链上的资产管理、合约调用等，需要通过 RPC转换中间件，将 FIBOS RPC节点提供的接口能力转换并封装成以太坊 RPC 节点接口格式并提供相同接口能力

_应用实例方案：_
Wormhole wallet
- 用户使用 Metamask 签名
- 为用户生成 FIBOS-Layer2 账号（可选特殊昵称或默认随机）
- 生成成功，用户即可使用以太坊账户交互任意 FIBOS DeFi 应用（需该应用支持），使用跨链充提或链上转账
- 每次交互均需 Metamask 签名

_迭代和交付规划_
Crypto Aboriginal 将在本提案通过后确定至少三个项目里程碑，并定期向 FIBOS 开发委员会提交代码以供代码安全和进度核验。

_申请金额及相关_
本提案为非盈利性质的基础设施类产品，根据现有工作量，结合开发成本(方案设计、密码学验证、开发实现、测试、文档等)，CrytoAboriginal(CA) 评估后拟申请：

 **FO 2,800,000** 

支付方式根据三个里程碑完成情况执行，分三期 50%、30%、20% 完成支付。

_交付物_ 

1. 可供 FIBOS 开发者社区使用的开源代码
2. 支持任意 ETH 同构链私钥借助 MetaMask 对 FIBOS 链上事务签名和操作
3. 3 个月内的 BUG 修复响应和核心问题迭代（如有）
4. 开放审计问题悬赏
5. 升级文档

**提案进程**
根据已有的 FIBOS 生态激励提案机制，我们将遵守以下规则完成提案：

申请期
A.小明认为 FIBOS 社区需要有项目 X，并在社区内与不同开发者或用户有过探讨。
B.小明撰写申请提案，内容包括：提案说明（什么样的产品或事务、对 FIBOS 生态的价值、具体的实现方案、小明的简介）和提案执行规划（时间节点、交付给社区的可验证的成果）
目前阶段👉C.小明联系【开发者委员会】，申请讨论
投票期
下一阶段👉D.开发者委员会认为方案可行，60% 以上成员在提案贴内明确回复可行后，小明可申请发起投票。
E.小明说明申请额度、拨付方案、投票周期、投票方式
F.【BP/基金会】按照【FO 持有者投票】的结果拨付 FO
拨付期
G.小明按期提交核验资料，得到【开发者委员会】核验通过后申请【BP/基金会】继续拨付
H.小明完成提案项目，交付代码并获得最终部分的 FO 激励

_提案方简介_
CrytoAboriginal(CA) 是由区块链行业早期布道投资者和持续创业者们共同发起的 DAO 形式区块链孵化投资机构。CrytoAboriginal(CA)擅长为初创社区、不同行业伙伴提供区块链产业解决方案、DeFi 产品设计和经济生态模型等方案、资源，并期待能够共同见证区块链驱动的信息价值社会到来。
CrytoAboriginal(CA) FIBOS 节点名称：cryptoaborig

**Comment by BIU64** [comment]
作为开发者委员会成员和提案发起方之一，我赞同该提案😆（如果有比该方案更好的也欢迎提交！）

**Reply by LEONG-BTC** [reply]
我赞同该提案😆

**Reply by william2041** [reply]
赞同，项目开始后，希望提案方能展示进度或里程，以相互监督，落实到位

**Reply by Sutarset** [reply]
没完全看懂，但支持😂

**Reply by BIU64** [reply]
> 喜爱，项目开始后，希望提案方能展示或引导，以相互监督，落实到位

+1 透明是必须的！所有项目都需要来说服委员会成员。如果是一个完全匿名的陌生团队，委员会和社区的要求肯定会更高。

**Reply by LEONG-BTC** [reply]
> > 喜爱，项目开始后，希望提案方能展示或引导，以相互监督，落实到位
> 
> +1 透明是必须的！所有项目都需要来说服委员会成员。如果是一个完全匿名的陌生团队，委员会和社区的要求肯定会更高。

希望尽快推进，创造无限可能

**Reply by kilmas** [reply]
agree

**Reply by kilmas** [reply]
技术论证和调研过了吗，实现阶段有详细的吗

**Reply by luoyhang003** [reply]
> 技术论证和调研过了吗，实现阶段有详细的吗

做了一轮基础的技术验证，实现的过程中肯定还有很多坑，遇一个排一个吧

**Reply by brighteyes99** [reply]
agree

**Comment by aSynch1889** [comment]
会让社区参与开发吗？

**Reply by BIU64** [reply]
【CrytoAboriginal(CA) FIBOS 节点名称：cryptoaborig 】作为 FIBOS 社区开发者之一，如果成功通过该提案，会和能够推进该提案的其他社区开发者积极沟通，一起 make fo great again！😆

**Comment by luoyhang003** [comment]
Approved. 赞成

**Comment by BIU64** [comment]
截至目前，本提案已通过了开发者委员会的 6 位委员支持，CrytoAboriginal(CA) 在此发起社区投票：
请 FO 持有者
**1.如果同意本提案，请向 cryptoaborig 转入 2 FO ，则视为该地址同意本提案。
2.如果不同意本提案，请向  cryptoaborig 转入 1，则视为该地址否决了本提案。**
3.投票期为_**【即时起至10.26 23:59 分(UTC+8)】**_。时间截至后，CrytoAboriginal(CA)  将会对所有投票用户的 FO 持有量（锁仓+流动）进行快照。

结果判定：
**_1.所有参与投票地址的总 FO 持有量超过（50,000,000.00 FO）视为有效投票
2.投票中明确表达同意的票数（向 cryptoaborig 转入 2 FO 的地址的 FO 持有量）超过 51%（25,500,000 FO）_**
则本提案正式通过，CrytoAboriginal(CA)  将以链上数据和本提案全流程资料为依据申请 FO 拨付，并进入下一环节。

-------------------------

参考资料：FIBOS 社区提案《新项目提案激励流程及激励规则》【https://github.com/FIBOSIO/fibos/discussions/23】
申请期
A.小明认为 FIBOS 社区需要有项目 X，并在社区内与不同开发者或用户有过探讨。
B.小明撰写申请提案，内容包括：提案说明（什么样的产品或事务、对 FIBOS 生态的价值、具体的实现方案、小明的简介）和提案执行规划（时间节点、交付给社区的可验证的成果）
C.小明联系【开发者委员会】，申请讨论
投票期
D.开发者委员会认为方案可行，60% 以上成员在提案贴内明确回复可行后，小明可申请发起投票。
目前阶段👉E.小明说明申请额度、拨付方案、投票周期、投票方式
下一阶段👉F.【BP/基金会】按照【FO 持有者投票】的结果拨付 FO
拨付期
G.小明按期提交核验资料，得到【开发者委员会】核验通过后申请【BP/基金会】继续拨付
H.小明完成提案项目，交付代码并获得最终部分的 FO 激励


4.其他细节
激励额度：建议按周期设定激励总额，由不同项目自行博弈和争夺。
激励发放：由提案者根据实际需求拟定，但至少分为 3 期，比例不超过 5、3、2，分期支付，以确保激励有效性
关于投票：
A.单一提案投票中 FO 持有者投票额至少超过流通量 5% 才能视为有效投票（50,000,000）
B.投票方式需合理合规无漏洞，否则【BP/基金会】可拒绝拨付
C.投票中明确表达同意的票数超过 51%，该提案才能被认为有效
D.目前尚无有效的社区投票工具，现提出一可用方案：a.根据 FO持有数量（Stake+流动）计算权重 b.任意地址在三天内（解锁期限）向特定地址中转入特定数量的 FO 完成投票 c.当投票期截至时快照，并根据快照时不同地址持有 FO 数量确定票权。
E.如某提案具有可实现属性，但报价过高导致未被投票通过，则可重新申请投票。但其多次重复申请必然会导致【开发者委员会】或 FO 持有用户疲烦，从而形成报价和投票的博弈。

**Comment by BIU64** [comment]
截止 2021-10-26 23:59(UTC+8) ，块高度 197036465
符合本提案规则的投票地址共计 38 个，投票截至时总投票用户有效余额为 67,901,490.45FO。

- [满足] 所有参与投票地址的总 FO 持有量超过（50,000,000.00 FO）视为有效投票
- [满足] 投票中明确表达同意的票数（向 cryptoaborig 转入 2 FO 的地址的 FO 持有量）超过 51%（25,500,000 FO）

**_因此，本提案将被视为有效并由 cryptoaborig 节点推进执行。_**

CrytoAboriginal(CA) 将按照以下流程开始推进

1. 申请拨付首批 fo（拨付规则按照上述提案内容执行）
2. 投入资源开始进行开发推进

-----------------------------------

本提案投票规则明确，数据可于 FIBOS 区块链上无门槛公开验证：https://see.fo/accounts/cryptoaborig
验证方法：
1. 部署 FIBOS 网络同步节点或使用其他区块链浏览器，获得本提案投票地址（在规定的投票期内，按照投票规则向投票地址执行交易的地址）
2. 将上述投票地址在验证时区块至 197036465 区块间交易剔除，以得到 197036465 时投票地址余额
3. 与本提案公开数据核对，确认无误

投票地址为<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

blockchain33
--
dongcheno5o2
hellojianhui
fibosfoniubb
zwdd1234dong
dong1234shao
xiao1234dong
plasmidfibos
xuanyuan1122
zhutaiyu1111
1b51kv3vzemg
liangdefibos
brighteyes11
3b3oksxcm1y4
fibosrockskr
chedongwei12
cnsakxsg4dyg
eos133245eos
xhlcfn22kfek
dfghjkl12345
lekvgggtgm3i
fibosbankbtc
pandafibosio
fibos3333221
hhhhhhqqqqqq
onlyeosfibos
leavemealone
defi1234defi
dictionary12
12345to54321
fiboswilliam
hcaoitygzjpi
fibosfibosoo
mxz333333333
huazi12345zz
liang3333333
biutothemoon
chenchafodev


</body>
</html>

查看链接：
 https://see.fo/accounts/cryptoaborig

---------------------------------

参考资料：FIBOS 社区提案《新项目提案激励流程及激励规则》【https://github.com/FIBOSIO/fibos/discussions/23】
申请期
A.小明认为 FIBOS 社区需要有项目 X，并在社区内与不同开发者或用户有过探讨。
B.小明撰写申请提案，内容包括：提案说明（什么样的产品或事务、对 FIBOS 生态的价值、具体的实现方案、小明的简介）和提案执行规划（时间节点、交付给社区的可验证的成果）
C.小明联系【开发者委员会】，申请讨论
投票期
D.开发者委员会认为方案可行，60% 以上成员在提案贴内明确回复可行后，小明可申请发起投票。
E.小明说明申请额度、拨付方案、投票周期、投票方式
目前阶段👉F.【BP/基金会】按照【FO 持有者投票】的结果拨付 FO
拨付期
下一阶段👉G.小明按期提交核验资料，得到【开发者委员会】核验通过后申请【BP/基金会】继续拨付
H.小明完成提案项目，交付代码并获得最终部分的 FO 激励

**Reply by william2041** [reply]
excellent，梦回2018.8.28，加油

**Reply by LEONG-BTC** [reply]
社区首个提案 顺利通过，下一步 ，星辰大海

**Comment by BIU64** [comment]
经过沟通确认后，本投票结果已得到认可和确认，目前为等待首批拨付到账阶段。
CA 已生成 wormholefofo 地址用于接收本提案激励。

**_wormholefofo_**

在首批激励发放后，我们将
1.开发规划（包括项目里程碑、开发周期和各对应预估时间节点）
2.发布技术方案

欢迎其他开发者或社区成员监督项目具体推进进度，以及提交任何技术或其他方面建议。

----------------------------------------------------------

参考资料：FIBOS 社区提案《新项目提案激励流程及激励规则》【https://github.com/FIBOSIO/fibos/discussions/23】
申请期
A.小明认为 FIBOS 社区需要有项目 X，并在社区内与不同开发者或用户有过探讨。
B.小明撰写申请提案，内容包括：提案说明（什么样的产品或事务、对 FIBOS 生态的价值、具体的实现方案、小明的简介）和提案执行规划（时间节点、交付给社区的可验证的成果）
C.小明联系【开发者委员会】，申请讨论
投票期
D.开发者委员会认为方案可行，60% 以上成员在提案贴内明确回复可行后，小明可申请发起投票。
E.小明说明申请额度、拨付方案、投票周期、投票方式
目前阶段👉F.【BP/基金会】按照【FO 持有者投票】的结果拨付 FO
拨付期
下一阶段👉G.小明按期提交核验资料，得到【开发者委员会】核验通过后申请【BP/基金会】继续拨付
H.小明完成提案项目，交付代码并获得最终部分的 FO 激励

**Comment by BIU64** [comment]
在经过沟通确认后，CA 将先行发布技术实现路径和里程碑，并在预算拨付后正式启动开发（公示监督地址：wormholefofo）

为了实现 FIBOS-WARMHOLE 提案中的愿景，现将实现路径拆分为如下技术基础：
### **技术实现路径**
**1. 公钥以及数字签名转录算法实现**

该部分内容为整个提案中技术的关键，要实现能够使用MetaMask插件进行FIBOS链上的资产管理、合约调用等，需要支持将Ethereum账号体系以及Ethereum密码学体系中的公钥（地址）以及数字签名转换为FIBOS可识别、可验证的格式。同时，在这一转录过程中不应额外使用用户的私钥，应当再不接触用户私钥的过程中，安全完成这一转换。这一技术需要密码学的知识，以及通过大数计算等数学方法解决这一技术难题。

**2. 转录算法服务化**

通过将实现的Ethereum-FIBOS公钥以及签名转录算法进行服务化，为社区以及其他开发者提供稳定的转录服务。社区开发者可以使用这一服务通过调用MetaMask的Native签名算法来实现对FIBOS链上的资产进行管理以及实现合约调用（需开发者开发的智能合约符合协议标准）

**3. 算法SDK化**

CA（cryptoaborig 节点） 将会把已实现的转录算法封装成必要编程语言版本的SDK，供各个社区开发者灵活使用。其中包括原始基础算法的调用以及依赖算法服务组件的RPC调用等，方便开发者灵活配置与使用，实现自由组合实现等。

**4. 项目完全开源**

当项目开发完成并且稳定运行后，CA（cryptoaborig 节点）将会依照国际主流开源标准，将项目代码以及文档等开源至GitHub，并且欢迎其他开发者的Bug Submit以及二次开发等。

### **关键里程碑**

- 实现公钥以及数字签名转录算法 - 2022/01/20
- 转录算法服务化 - 2022/02/05
- 算法SDK发布 - 2022/02/18
- 完成项目开源 - 2022/03/05

--------------------------------
**### 附：**
在 FIBOS-WARMHOLE(https://github.com/FIBOSIO/fibos/discussions/25) 提案中，CA（cryptoaborig 节点） 仅提出了实现签名层面的转化和相关功能的 SDK 封装任务。在此基础上，FIBOS-WARMHOLE 的 SDK 本质上只是从技术上支持了该功能，仍然需要第三方应用（具有支持以太坊私钥管理 FIBOS 账户需求）的接入和使用才能真正让  FIBOS-WARMHOLE 的价值被展现。

但在经过频繁的内部讨论后，CA 认为：

- 这样的工具等待社区其他项目采用的时效无法确认
- CA 再次提起一个应用了该 SDK 的提案可能也会花费更长的时间成本，而我们期待耗费了 FIBOS 生态资源的工具可以真正的更早的被应用起来。

因此，在 FIBOS-WARMHOLE 提案的任务内容之外，CA（cryptoaborig 节点）将调用自身资源，无偿开发使用该提案的示例应用：_**Warmhole WEB Wallet**_。这样的产品由 SDK 本身的开发团队-CA 来实现相对也更加高效。

目前，该产品的形态包括：

- 一个独立的 WEB 页，支持:
  - 使用 MetaMask 登录，创建由该 ETH 地址 A 管理的 FIBOS 账户 A-FIBOS。
  - 支持由该 ETH 对其对应的 A-FIBOS 账户中的资产在 WEB 中管理，包括转账、stake 等常规操作。

- 一个用于为 WEB 页用户提供 FIBOS 账号免费注册服务的专用 FIBOS GAS 付费账户（预存储费用将由 CA 自行承担,消耗速度与 MetaMask-ETH 用户的 FIBOS 账户注册量有关）

- 相关的文档，将主要包括以下内容（中文版本优先，英文版欢迎社区提交志愿版本）：
  - 什么是 FIBOS-WARMHOLE SDK
  - 什么是 Warmhole WEB Wallet
  - 你可以使用 Warmhole WEB Wallet 做些什么？
  - 给其他有意向采用  FIBOS-WARMHOLE SDK&Warmhole WEB Wallet 的开发者的建议。

其他：
- 本产品开发将尽量由 FIBOS-WARMHOLE 剩余预算覆盖，差额部分由 CA（cryptoaborig 节点） 自行承担，不再发起提案。
- 所有代码将无偿开源，供有需求的其他 FIBOS 社区开发者采用，尽早将 FIBOS 的 ETH Layer2 价值展现。
- 该项目的开发周期目前预估将在稍微延后的基础上整体与 SDK 同步，但视实际情况可能会有所调整。

加油，Fiboser ⛽⛽

**Comment by BIU64** [comment]
**【重要更新】**
本提案因流程问题，之前规划的里程碑都基本自动超时；CA 团队资源都已先排给其他事务。
从对提案负责任的考虑，如本提案一旦正式开始，CA 将要对完成度和完成质量确保，但在内部评估后， CA 无法确保仍有能力和资源完成预期效果，为避免影响开发进度，在沟通和确认后：
**CA 在此正式撤回本提案。**

关于撤回流程：
本提案尚未正式拨付原计划金额，因此不涉及资产退回流程。
CA 欢迎其他社区团队参考已有提案内容，重新发起提案。

在本条更新发布后：
**请勿继续向 （公示监督地址：wormholefofo）执行首批预算拨付工作。**
**由 CA 提出的原有开发规划和内容已作废。**

上述提案内容无偿供其他有意向继续推进的社区开发者参考推进，具体进度和相关激励 FO 等均由新的提案者于其提案中公布，与  CA（cryptoaborig 节点） 无关。

如其他开发者在开发过程中有需要协助的工作内容，请向我们留言，将会尽力协助。

此致，感谢一直以来关注本提案的社区成员，也同时为无法完成感到抱歉。在后续时机成熟，且 CA 具有更充分准备时，将积极发起其他提案继续参与 FIBOS 社区建设。

**Reply by luoyhang003** [reply]
欢迎社区中其他对此提案感兴趣的开发者个人或者团队一起交流。

**Comment by mosquitohater2** [comment]
若有其他开发者个人或团队对此提案感兴趣或有更好的方案，欢迎发起新提案，FIBOS 开源委员会表示全力支持。

**Comment by charlesZ125** [comment]
请问何时开源呢？
